src/streaming-client.ts

```diff
@@ -0,0 +1,90 @@
+import * as fs from 'fs';
+import * as sockets from 'websocket';
+import { Duplex, PassThrough } from 'stream';
+
+import AudioConfig from './models/streaming/AudioConfig';
+import BufferedDuplex from './models/streaming/BufferedDuplex';
+import StreamingResponse from './models/streaming/StreamingResponse';
+
+export class RevAiStreamingClient {
+    accessToken: string;
+    config: AudioConfig;
+    baseUrl: string;
+    requests: PassThrough;
+    responses: PassThrough;
+    client: sockets.WebSocketClient;
+    constructor(accessToken: string, config: AudioConfig, version = 'v1alpha'){
+        this.accessToken = accessToken;
+        this.config = config;
+        this.baseUrl = `wss://api.rev.ai/speechtotext/${version}/stream?` +
+            `access_token=${accessToken}` +
+            `&content_type=${config.getContentTypeString()}`;
+        this.requests = new PassThrough();
+        this.responses = new PassThrough({ objectMode: true });
+        this.client = new sockets.client();
+    }
+
+    start (): Duplex {
+        var self = this;
+        this.client.on('connectFailed', function(error: Error) {
+            console.log("Failed to connect to streaming service");
+            self.requests.end();
+            throw error;
+        });
+        this.client.on('connect', function(connection: any) {
+            connection.on('error', function(error : Error) {
+                console.log("Error thrown in connection to streaming service");
+                self.requests.end();
+                self.responses.push(null);
+                throw error;
+            });
+            connection.on('close', function(code: number, reason: string) {
+                console.log(`Connection closed, ${code}: ${reason}`);
```

Comment:
I'm debating what the best way to return this information to the user is. I have 3 potential solutions
1. RevAiStreamingClient inherits from WebSocketClient so all these events can be thrown and seen by the user.
2. Store close status as a property
3. Write a close message to the output stream.

I personally think #1 is the best since it gives the user the most flexibility but I wanted to hear other opinions